projects/step-core/src/lib/modules/tree/services/tree-state.service.ts

```diff
@@ -320,15 +336,26 @@ export class TreeStateService<T, N extends TreeNode> implements OnDestroy {
 
   addChildrenToSelectedNode(...children: T[]): void {
     const parentId = this.selectedInsertionParentId$.value;
+
     if (!parentId) {
       return;
     }
 
     const newNodes = children.map((child) => this._treeNodeUtils.convertItem(child));
+
     if (!this.isPossibleToInsert(parentId, ...newNodes)) {
       return;
     }
 
+    const parent = this.findNodeById(parentId);
+
+    if (parent) {
+      newNodes.forEach((node) => {
+        node.isSkipped = parent.isSkipped;
+        (node as unknown as { originalArtefact: AbstractArtefact }).originalArtefact.skipNode!.value = parent.isSkipped;
```

Comment:
TreeStateService is abstract common thing. In theory it should know nothing about node's content.

Placing this logic here not a clean solution.
I would prefer to move somewhere else, where we have already know that we are working with AbstractArtefact